docs/How_To/Tutorials/document_reader.rst

```diff
@@ -0,0 +1,253 @@
+
+4. `agent_config`: This is the configuration for the agent. The default configuration is used in this tutorial.
+
+5. `num_runs`: This is the number of runs the agent will execute an action. In this tutorial, the agent will execute only once.
+
+6. `actions`: This is the list of actions that the agent can execute. In this case, the agent can execute the `doc_search` action.
+
+
+Put it all together
+********************
+
+Now, we can put everything together to create the PDF reader. Create a file called `main.py` and add the following code:
+
+.. code-block:: python
+
+    from argparse import ArgumentParser
+
+    from hydra.utils import instantiate
+    from omegaconf import OmegaConf
+
+    from sherpa_ai.agents import QAAgent
+    from sherpa_ai.events import EventType
+
+
+    def get_qa_agent_from_config_file(
+        config_path: str,
+    ) -> QAAgent:
+        """
+        Create a QAAgent from a config file.
+
+        Args:
+            config_path: Path to the config file
+
+        Returns:
+            QAAgent: A QAAgent instance
+        """
+
+        config = OmegaConf.load(config_path)
+
+        agent_config = instantiate(config.agent_config)
+        qa_agent: QAAgent = instantiate(config.qa_agent, agent_config=agent_config)
+
+        return qa_agent
+
+    if __name__ == "__main__":
+        parser = ArgumentParser()
+        parser.add_argument("--config", type=str, default="agent_config.yaml")
+        args = parser.parse_args()
+
+        qa_agent = get_qa_agent_from_config_file(args.config)
+
+        while True:
+            question = input("Ask me a question: ")
+
+            # Add the question to the shared memory. By default, the agent will take the last
+            # message in the shared memory as the task.
+            qa_agent.shared_memory.add(EventType.task, "human", question)
+            result = qa_agent.run()
+            print(result)
+
+In this code, we define a function `get_qa_agent_from_config_file` that reads the configuration file and creates a QAAgent instance. We then create a QAAgent instance using the `get_qa_agent_from_config_file` function and run the agent in a loop. The agent will ask for a question and then answer the question based on the content of the PDF file.
+
+
+Run the PDF reader
+******************
+
+Before we can run the PDF reader, we need to add a environment variable for OpenAI API key. You can get the API key from the OpenAI website. Create a file called `.env` and add the following code:
+
+.. code-block:: bash
+
+    OPENAI_API_KEY=<YOUR_API_KEY>
+
+Now, you can run the PDF reader by running the following command:
+
+.. code-block:: bash
+
+    python main.py --config agent_config.yml
+
+You will be prompted to ask a question. You can ask any question about the content of the PDF file. The agent will then answer the question based on the content of the PDF file.
+
+.. image:: imgs/pdf_reader.png
```

Comment:
this image is recorded on a very while screen, it makes it hard to read it. could you replace it with one from a narrower screen to improve readability?